### DISCUSSION TITLE
ESLint flat config extends doesn't replicate `.eslintrc.js` extends

### DISCUSSION THREAD

**Discussion by matthew-dean** [eslint/eslint#17477] [General]
One of the things that you can trivially do in `.eslintrc.js` files is extend per override. For example:
```js
module.exports = {
  extends: ['standard'],
  overrides: [
    {
      files: ['*.ts', '*.tsx', '*.vue'],
      extends: ['standard-with-typescript']
    }
  ]
}
```
The above, for example, can be used to lint `.js` files with "eslint-config-standard" and `.ts` files with "eslint-config-standard-with-typescript".

However, the examples for the flat config only seem to support extending the root. As in:
```js
const standard = require('eslint-config-standard')

const config = [
  standard,
  {}
]
```
How does one apply extends to only a matching glob?

**Comment by Rec0iL99** [comment]
Hi @matthew-dean, you can do something like this:

```js
import standard from "eslint-config-standard";
import standardTs from "eslint-config-standard-with-typescript"

export default [
  {
    files: ["*.js"],
    ...standard,
  },
  {
    files: ["*.ts"],
    ...standardTs,
  },
];
```

Here the config from `eslint-config-standard` only apply to `js` files, while the config from `eslint-config-standard-with-typescript` applies to `ts` files alone.

[Docs](https://eslint.org/docs/latest/use/configure/configuration-files-new#using-predefined-configurations)
